src/weather_type.cpp

```diff
@@ -209,7 +209,12 @@ void weather_type::load( const JsonObject &jo, const std::string & )
         optional( weather_effect_jo, was_loaded, "trait_id_to_add", effect.trait_id_to_add );
         optional( weather_effect_jo, was_loaded, "trait_id_to_remove", effect.trait_id_to_remove );
         optional( weather_effect_jo, was_loaded, "target_part", effect.target_part );
-        optional( weather_effect_jo, was_loaded, "damage", effect.damage, 0 );
+        // TODO: make this work with `was_loaded`
+        if( weather_effect_jo.has_array( "damage_data" ) ) {
+            effect.damage_data = load_damage_instance( weather_effect_jo.get_array( "damage_data" ) );
+        } else if( weather_effect_jo.has_object( "damage_data" ) ) {
+            effect.damage_data = load_damage_instance( jo );
+        }
```

Comment:
I believe you can use `assign()` here, like items do.

Reply:
Thanks.